### DISCUSSION TITLE
How can I get type support on withRouter props?

### DISCUSSION THREAD

**Discussion by sean256** [vercel/next.js#18171] [Help] [answered]
I'm fairly new to TypeScript so apologies if this is super simple.

So this is what I have:

```js
import { withRouter } from 'next/router';

function TestComp({ router }) {
	const { pathname } = router; // how can I get type support on router?
	return (
		<div>{pathname}</div>
	);
}

export default withRouter(TestComp);
```

How can I get TypeScript type support on the router prop?

**Comment by jamesmosier** [comment, marked as answer]
Based on a couple [issues](https://github.com/vercel/next.js/issues/7311), it appears it is not exported via `next/router` although it is via [this workaround](https://github.com/vercel/next.js/issues/9424#issuecomment-554539575): `import { WithRouterProps } from 'next/dist/client/with-router';` although that is referencing internal library code that is subject to change.

You might be able to use `import { NextRouter } from 'next/router';` which has all the router types as well.
